x/payment/client/cli/wirepayformessage.go

```diff
@@ -52,7 +53,17 @@ func CmdWirePayForMessage() *cobra.Command {
 			}
 
 			// create the MsgPayForMessage
-			pfmMsg, err := types.NewWirePayForMessage(namespace, message, consts.MaxSquareSize)
+			squareSizes, err := cmd.Flags().GetUintSlice(cflags.FlagSquareSizes)
+			if err != nil {
+				return err
+			}
+
+			squareSizes64 := make([]uint64, len(squareSizes))
+			for i := range squareSizes {
+				squareSizes64[i] = uint64(squareSizes[i])
+			}
```

Comment:
The Flags() only has the option to get an []uint. We need to convert it to []uint64 to pass it to `NewWirePayForMessage`.
This is ugly, do we have a better way to do this?

Reply:
We can just create a `parseSquareSizes` func or something. No other real way around it, I don't think

> Do we need to add the condition to check whether the square size is a power of 2?

yes! good thinking. Now that user can enter square size, we will need extra protections to ensure that the square size is a power of 2. Can we actually add that in the [`NewWirePayForMessage`](https://github.com/celestiaorg/celestia-app/blob/63519eccc053e9695af5901164dfd5da5fe3fb24/x/payment/types/wirepayformessage.go#L31-L32) function? Then we should also check for that in the `ValidateBasic` method of `MsgWirePayForMessage`

thanks!